python/foxglove-sdk/src/websocket_server.rs

```diff
@@ -473,11 +483,42 @@ impl From<PyCapability> for foxglove::websocket::Capability {
     }
 }
 
+/// An asset server for live visualization.
+///
+/// The handler must be a callback function which takes the uri as its argument, and returns the
+/// asset `bytes`. If the handler returns `None`, a "not found" message will be sent to the client.
+/// If the handler raises an exception, the stringified exception message will be returned to the
+/// client as an error.
+pub struct AssetServer {
```

Comment:
I think this could be simpler if you structured it as a closure passed as the blocking callback (if we're not using the builder you'd need to construct it yourself to set it in server options.)

At least it does the spawning and arc for you.

Reply:
I'm struggling to come up with the conversion for that. I'll merge this for now and take another look when I flesh out the URDF asset example.